narwhal/primary/src/tests/primary_tests.rs

```diff
@@ -0,0 +1,258 @@
+            PrimaryChannelMetrics::NAME_NEW_CERTS,
+            PrimaryChannelMetrics::DESC_NEW_CERTS,
+        )
+        .unwrap(),
+    );
+    let (tx_feedback_2, rx_feedback_2) = types::metered_channel::channel(
+        CHANNEL_CAPACITY,
+        &prometheus::IntGauge::new(
+            PrimaryChannelMetrics::NAME_COMMITTED_CERTS,
+            PrimaryChannelMetrics::DESC_COMMITTED_CERTS,
+        )
+        .unwrap(),
+    );
+    let (tx_get_block_commands_2, rx_get_block_commands_2) = types::metered_channel::channel(
+        1,
+        &prometheus::IntGauge::new(
+            PrimaryChannelMetrics::NAME_GET_BLOCK_COMMANDS,
+            PrimaryChannelMetrics::DESC_GET_BLOCK_COMMANDS,
+        )
+        .unwrap(),
+    );
+    let initial_committee = ReconfigureNotification::NewEpoch(committee.clone());
+    let (tx_reconfigure_2, _rx_reconfigure_2) = watch::channel(initial_committee);
+    let consensus_metrics = Arc::new(ConsensusMetrics::new(&Registry::new()));
+
+    // println!("Primary 2 Information");
+    // dbg!(authority_2.network_keypair().copy().public());
+
+    // Spawn Primary 2
+    Primary::spawn(
+        name_2.clone(),
+        signer_2,
+        authority_2.network_keypair().copy(),
+        Arc::new(ArcSwap::from_pointee(committee.clone())),
+        worker_cache.clone(),
+        primary_2_parameters.clone(),
+        store.header_store.clone(),
+        store.certificate_store.clone(),
+        store.payload_store.clone(),
+        store.vote_digest_store.clone(),
+        /* tx_consensus */ tx_new_certificates_2,
+        /* rx_consensus */ rx_feedback_2,
+        /* dag */
+        tx_get_block_commands_2,
+        rx_get_block_commands_2,
+        Some(Arc::new(
+            Dag::new(&committee, rx_new_certificates_2, consensus_metrics).1,
+        )),
+        NetworkModel::Asynchronous,
+        tx_reconfigure_2,
+        tx_feedback_2,
+        &Registry::new(),
+        None,
+    );
+
+    // Wait for tasks to start
+    tokio::time::sleep(Duration::from_secs(1)).await;
+
+    // Test getting all connected peers for primary 1
+    let resp = reqwest::get(format!(
+        "http://127.0.0.1:{}/peers",
+        primary_1_parameters.network_admin_server_port
+    ))
+    .await
+    .unwrap()
+    .json::<Vec<String>>()
+    .await
+    .unwrap();
+
+    // println!("Primary 1 Connected Peers");
+    // dbg!(resp.clone());
+
+    // Assert we returned 2 peers (1 other primary spawned + 1 worker spawned)
+    assert_eq!(2, resp.len());
+
+    // Test getting all connected peers for primary 2
+    let resp = reqwest::get(format!(
+        "http://127.0.0.1:{}/peers",
+        primary_2_parameters.network_admin_server_port
+    ))
+    .await
+    .unwrap()
+    .json::<Vec<String>>()
+    .await
+    .unwrap();
+
+    // println!("Primary 2 Connected Peers");
+    // dbg!(resp.clone());
+
+    // Assert we returned 2 peers (1 other primary spawned + 1 other worker)
+    assert_eq!(2, resp.len());
```

Comment:
Do you think it would be worth it to assert the actual PeerIds of the peers we're connected with?